cmd/trtlsim/main.go

```diff
@@ -0,0 +1,238 @@
+			return nil, err
+		}
+
+		var certPool *x509.CertPool
+		if certPool, err = pool.GetCertPool(false); err != nil {
+			return nil, err
+		}
+		var u *url.URL
+		if u, err = url.Parse(s.Endpoint); err != nil {
+			return nil, err
+		}
+		conf := &tls.Config{
+			ServerName:   u.Host,
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      certPool,
+		}
+		opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(conf)))
+	}
+
+	// Connect the replica client
+	var cc *grpc.ClientConn
+	if cc, err = grpc.Dial(s.Endpoint, opts...); err != nil {
+		return nil, err
+	}
+	return pb.NewTrtlClient(cc), nil
+}
+
+// Create a fixed number of random accesses across the namespace and keyspace options
+func (s *Simulator) accessor(client pb.TrtlClient) {
+	// Make as many accesses as directed by the global variable/config
+	for a := 1; a <= accesses; a++ {
+
+		// randomly select a namespace from namespaces
+		ns := namespaces[rand.Intn(len(namespaces))]
+
+		// randomly select a key from keyspace and cast to bytes
+		key := []byte(strconv.Itoa(rand.Intn(keyspace)))
+
+		// select from read, write, delete using probabilities
+		switch s.Selector.Pick().(string) {
+		case "read":
+			// execute Get
+			req := &pb.GetRequest{
+				Key:       key,
+				Namespace: ns,
+				Options: &pb.Options{
+					ReturnMeta: false,
+				},
+			}
+			if _, err := client.Get(context.TODO(), req); err != nil {
+				// Skip if the object doesn't exist
+				if serr, ok := status.FromError(err); ok && serr.Code() == codes.NotFound {
+					continue
+				}
+				panic(fmt.Errorf("could not read from database: %v", err))
+			}
+
+		case "write":
+			// create random data
+			val := make([]byte, chunkSize)
+
+			// execute Put
+			req := &pb.PutRequest{
+				Key:       key,
+				Value:     val,
+				Namespace: ns,
+				Options: &pb.Options{
+					ReturnMeta: false,
+				},
+			}
+			if _, err := client.Put(context.TODO(), req); err != nil {
+				fmt.Print(err)
+				panic("could not write to database")
+			}
+		case "delete":
+			// First check that the object exists
+			req := &pb.GetRequest{
+				Key:       key,
+				Namespace: ns,
+				Options: &pb.Options{
+					ReturnMeta: false,
+				},
+			}
+			// Object doesn't exist so skip this one
+			if _, err := client.Get(context.TODO(), req); err != nil {
+				if serr, ok := status.FromError(err); ok && serr.Code() == codes.NotFound {
+					continue
+				}
+			} else {
+				// Object does exist so execute Delete
+				req := &pb.DeleteRequest{
+					Key:       key,
+					Namespace: ns,
+					Options: &pb.Options{
+						ReturnMeta: false,
+					},
+				}
+				if _, err := client.Delete(context.TODO(), req); err != nil {
+					if serr, ok := status.FromError(err); ok && serr.Code() == codes.NotFound {
+						continue
+					}
+					panic(fmt.Errorf("could not delete from database: %v", err))
+				}
+			}
+		default:
+			panic(errors.New("unknown database operation"))
+		}
+	}
```

Comment:
At the end of this function, just for debugging and analysis would it be possible to print the number of gets, puts, and deletes that the accessor just performed using `log.Println`? That will help us see that the sim is running when we deploy it and will give us a sense of what accesses happened in the last round.